### PR TITLE
feat(filestore): Add GCS backend

### PR DESCRIPTION
**Pull request by mattrobenolt**
This has been modified from django-storages to work for us. I'm doing some more investigation to make sure this is the most optimized way since it seemed a bit verbose in HTTP requests.

### CODE REVIEW

**Comment by getsentry-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">You should update CHANGES due to the size of this PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

**Comment by nlopes** [comment]
I'd love to see this implemented. @mattrobenolt any chance this will get picked up? I can help if needed.

**Review by joshuarli — DISMISSED** [review]

**Comment by JoshuaRLi** [comment]
Good stuff Matt.

File: docs/filestore.rst [hunk 1]
```diff
@@ -17,7 +17,22 @@ File System Backend
       location: '/tmp/sentry-files'
 
 
-S3 Backend
+Google Cloud Storage Backend
+----------
+
+In addition to the configuration below, you'll need to make sure the shell
```
[the remark this answers is not in the document]
Comment (jody-frankowski) [reply docs/filestore.rst]: According to [this documentation](https://cloud.google.com/storage/docs/reference/libraries#setting_up_authentication), we should export an environment variable pointing to the service account key file (or let sentry do that internally?).
You should be able to create a service account key [like this](https://cloud.google.com/iam/docs/creating-managing-service-account-keys#creating_service_account_keys).

[the remark this answers is not in the document]
Comment (mattrobenolt) [reply docs/filestore.rst]: Yeah, this is one way, or if your servers are running within Google Compute, you can assign permissions to the servers themselves, so authentication is implicit.

File: docs/filestore.rst [hunk 2]
```diff
@@ -17,7 +17,22 @@ File System Backend
       location: '/tmp/sentry-files'
 
 
-S3 Backend
+Google Cloud Storage Backend
+----------
```
[the remark this answers is not in the document]
Comment (JoshuaRLi) [reply docs/filestore.rst]: Thanks Matt.

File: docs/filestore.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (JoshuaRLi) [reply docs/filestore.rst]: @mattrobenolt Are you just making a note here, or actually requesting that the instructions be changed to using `GOOGLE_APPLICATION_CREDENTIALS`?

[the remark this answers is not in the document]
Comment (mattrobenolt) [reply docs/filestore.rst]: Yeah, I'd argue that if we're adding it, we should add correct instructions.

[the remark this answers is not in the document]
Comment (JoshuaRLi) [reply docs/filestore.rst]: 24e7c4ed0c4e6860d6ce17bd90771fe615a1ad9d

**Pull request closed without merging**